cornucopia/contrast.py

```diff
@@ -184,7 +184,7 @@ def make_final(self, x, max_depth=float('inf')):
 
         vmin, vmax = x.min(), x.max()
         edges = torch.linspace(vmin, vmax, self.nk+1)
-        new_mu = torch.rand(self.nk) * (vmax - vmin) + vmin
+        new_mu = torch.rand(self.nk).to(x.device) * (vmax - vmin) + vmin
```

Comment:
Can you replace `.to(x.device)` with `.to(x)` ? This way it will also take care of the datatype :)

Reply:
Ah, excellent idea. I did not know something like that was possible!